where-filter.ts

```diff
@@ -206,14 +209,21 @@ function test(example, value) {
 	);
 }
 
-function whereFilter(where) {
-	if (typeof where === 'function') {
-		return where;
+export type FilterFunction<T> = (obj: T) => boolean;
+
+// TODO: real types
+// @ts-ignore
+export type Where<T> = any;
+
+export function whereFilter<T>(filter: Where<T> | FilterFunction<T>): FilterFunction<T> {
```

Comment:
I think this will allow you to remove the extra `export default ...` at the bottom
```suggestion
export default function whereFilter<T>(filter: Where<T> | FilterFunction<T>): FilterFunction<T> {
```

Reply:
yep, good point, updated